foundation/build.gradle.kts

```diff
@@ -21,6 +21,14 @@ android {
     buildFeatures {
         compose = true
     }
+
+    kotlin {
+        compilerOptions {
+            // From Kotlin 2.2, need to specify default rule for annotations
+            // See http://youtrack.jetbrains.com/issue/KT-73255
+            freeCompilerArgs.add("-Xannotation-default-target=param-property")
```

Comment:
We can move this into `library.gradle.kts` with other compiler arguments in order to factorize this in case another module needs it in the future.